### ISSUE TITLE
Device detection flags have mismatched types

### ISSUE THREAD

**Issue by krojew** [melonjs/melonJS#860]
me.device defines a set of flags for each possible device. According to the docs those are either true or false. In practice, because nobody bothered to add " !== null" in _detectDevice(), those flags can be either boolean or an array of strings. This might cause problems, and surely will cause confusion if someone prints it out.

**Comment by obiot** [comment]
i'm not sure i'm following you, if `null` or `undefined` is returned, then the flag value will be set to false, thanks to the ` | false` at the end.

for example :
```javascript
me.device.iOS = me.device.ua.match(/iPhone|iPad|iPod/i) || false;
````\

**Comment by krojew** [comment]
Yes, but note what will happen when match doesn't return null or undefined.

**Comment by obiot** [comment]
oh ok, my bad, I see your point, it should indeed be :
```javascript
me.device.iOS = typeof (me.device.ua.match(/iPhone|iPad|iPod/i)) !== "null";
```

**Comment by obiot** [comment]
that will be for tomorrow morning, off to bed here for now !

thank you for all your checks and reports again :P

**Comment by krojew** [comment]
Or simply use !== null without typeof, since the result is well defined.

**Comment by krojew** [comment]
Also, be ready for more reports since https://github.com/krojew/melonTS is in progress :)

**Comment by agmcleod** [comment]
@obiot can be done via test(), which returns a boolean. `me.device.iOS = /iPhone|iPad|iPod/i.test(me.device.ua)`

**Comment by obiot** [comment]
Indeed, even better than match that returns an array !
